velox/exec/prefixsort/PrefixSortEncoder.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cstdint>
+#include <functional>
+#include <memory>
+
+#include "velox/common/base/BitUtil.h"
+#include "velox/common/base/Exceptions.h"
+#include "velox/common/base/SimdUtil.h"
+
+namespace facebook::velox::exec::prefixsort {
+
+/// Provides encode/decode methods for PrefixSort.
+class PrefixSortEncoder {
+ private:
+  FOLLY_ALWAYS_INLINE static uint8_t flipSignBit(uint8_t byte) {
+    return byte ^ 128;
+  }
+
+ public:
+  template <typename T>
+  static FOLLY_ALWAYS_INLINE void encode(T value, char* row) {
+    VELOX_UNSUPPORTED("Cannot encode row from this type,");
```

Comment:
Just leave it unimplemented so any unsupported type will show up as compilation error instead of runtime exceptions

Reply:
If I leave it unimplemented, it`s ok in my dev env(ubuntu-linx , llvm clang17), but I will get a error in CI macos env :

`
/Users/distiller/project/./velox/exec/prefixsort/PrefixSortEncoder.h:40:35: error: inline function 'facebook::velox::exec::prefixsort::PrefixSortEncoder::decode<long>' is not defined [-Werror,-Wundefined-inline]
  static FOLLY_ALWAYS_INLINE void decode(char* row);
                                  ^
/Users/distiller/project/./velox/exec/prefixsort/PrefixSortEncoder.h:71:24: note: used here
    PrefixSortEncoder::decode<T>((char*)data.data() + i * sizeof(T));
`

ref : https://app.circleci.com/pipelines/github/facebookincubator/velox/40060/workflows/a9b81b88-3e54-4da5-905a-7b5c8f838882/jobs/273047


@Yuhta Is there some method to ' show up as compilation error instead of runtime exceptions'

Reply:
This is probably because `long` is different from `int64_t` on MacOS M1

Reply:
> This is probably because long is different from int64_t on MacOS M1

 I change `long` to `int64_t` in ut, and this method could work.， and  leave encode/decode mehtod unimplemented 

@Yuhta